sparkler-app/src/main/scala/edu/usc/irds/sparkler/pipeline/ParseFunction.scala

```diff
@@ -17,37 +17,57 @@
 
 package edu.usc.irds.sparkler.pipeline
 
-import java.io.ByteArrayInputStream
+import java.io.{InputStream, ByteArrayInputStream}
 
 import edu.usc.irds.sparkler.base.Loggable
-import edu.usc.irds.sparkler.model.CrawlData
-import org.apache.tika.metadata
+import edu.usc.irds.sparkler.model.{ParseData, CrawlData}
+import org.apache.commons.io.IOUtils
+import org.apache.tika.metadata.Metadata
 import org.apache.tika.parser.AutoDetectParser
-import org.apache.tika.sax.LinkContentHandler
+import org.apache.tika.sax.{WriteOutContentHandler, BodyContentHandler, LinkContentHandler}
 
 import scala.collection.JavaConverters._
 
 /**
   * Created by thammegr on 6/7/16.
   */
-object ParseFunction extends ((CrawlData) => Set[String]) with Serializable with Loggable {
+object ParseFunction extends ((CrawlData) => (ParseData)) with Serializable with Loggable {
 
-  override def apply(data: CrawlData): Set[String] = {
+  override def apply(data: CrawlData): (ParseData) = {
+    val parseData: ParseData = new ParseData()
+    var stream: InputStream = new ByteArrayInputStream(data.content.content)
+    val linkHandler = new LinkContentHandler()
+    val parser = new AutoDetectParser()
+    val meta = new Metadata()
+    val outHandler = new WriteOutContentHandler()
+    val contentHandler = new BodyContentHandler(outHandler)
+    LOG.info("PARSING  {}", data.content.url)
     try {
-      LOG.info("PARSING  {}", data.content.url)
-      val stream = new ByteArrayInputStream(data.content.content)
-      val linkHandler = new LinkContentHandler()
-      val parser = new AutoDetectParser()
-      val meta = new metadata.Metadata()
+      // Parse OutLinks
       meta.set("resourceName", data.content.url)
       parser.parse(stream, linkHandler, meta)
-      stream.close()
-      linkHandler.getLinks.asScala.map(_.getUri.trim).filter(!_.isEmpty).toSet
+      parseData.outlinks = linkHandler.getLinks.asScala.map(_.getUri.trim).filter(!_.isEmpty).toSet
     } catch {
       case e:Throwable =>
-        LOG.warn("PARSER-ERROR {}", data.content.url)
+        LOG.warn("PARSING-OUTLINKS-ERROR {}", data.content.url)
         LOG.warn(e.getMessage, e)
-        Set.empty[String]
+    } finally {
+      IOUtils.closeQuietly(stream)
+    }
+    try {
+      // Parse Text
+      stream = new ByteArrayInputStream(data.content.content)
+      parser.parse(stream, contentHandler, meta)
```

Comment:
@karanjeets 
FYI reusing `meta` object for two parse calls causes duplicate key:value insertions!
Fixing it.